fbw-common/src/systems/instruments/src/EFB/TroubleshootingContext.tsx

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2025 FlyByWire Simulations
+// SPDX-License-Identifier: GPL-3.0
+
+import { EventBus } from '@microsoft/msfs-sdk';
+import React, { useContext, useState } from 'react';
+import { TroubleshootingEvents } from '../../../shared/src/Troubleshooting';
+
+const TroubleshootingContext = React.createContext<string[]>(undefined as any);
+
+interface TroubleshootingContextProps {
+  eventBus: EventBus;
+}
+
+export const TroubleshootingContextProvider: React.FC<TroubleshootingContextProps> = ({ eventBus, children }) => {
+  const [errorLog, setErrorLog] = useState([] as string[]);
+
+  eventBus
```

Comment:
This will leak subscriptions every time it renders.

You should place this in a `useEffect` (with an empty dependency array) whose cleanup function destroys the subscription.